### PR TITLE
chore(permission): Adds minimum permission for operator

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <udit.gaurav@mayadata.io>

**What this PR does / why we need it**:

- This PR adds minimum permission to run chaosoperator
- An [e2e pipeline](https://gitlab.mayadata.io/litmuschaos/litmus-e2e/pipelines/12749) run with the given permission. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] Labelled this PR & related issue with `documentation` tag
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests

### CODE REVIEW

File: deploy/rbac.yaml [hunk 1]
```diff
@@ -18,9 +18,15 @@ metadata:
...
     name: litmus
 rules:
-- apiGroups: ["","apps","batch","litmuschaos.io","apps.openshift.io"]
-  resources: ["pods","jobs","deployments","replicationcontrollers","replicasets","daemonsets","statefulsets","deploymentconfigs","events","configmaps","services","secrets","chaosengines","chaosexperiments","chaosresults"]
-  verbs: ["get","create","update","patch","delete","list","watch","deletecollection"]
+- apiGroups: ["","apps","batch","apps.openshift.io"]
+  resources: ["jobs","deployments","replicationcontrollers","daemonsets","replicasets","statefulsets","deploymentconfigs","events","services","secrets"]
+  verbs: ["get","list","watch"]
+- apiGroups: ["","litmuschaos.io"]
+  resources: ["pods","configmaps","chaosengines","chaosexperiments","chaosresults"]
```
Comment (ksatchit) [inline deploy/rbac.yaml:25]: we need deletecollection of jobs as well (abort)?

File: deploy/rbac.yaml [hunk 2]
```diff
@@ -18,9 +18,15 @@ metadata:
   labels:
     name: litmus
 rules:
-- apiGroups: ["","apps","batch","litmuschaos.io","apps.openshift.io"]
-  resources: ["pods","jobs","deployments","replicationcontrollers","replicasets","daemonsets","statefulsets","deploymentconfigs","events","configmaps","services","secrets","chaosengines","chaosexperiments","chaosresults"]
-  verbs: ["get","create","update","patch","delete","list","watch","deletecollection"]
+- apiGroups: ["","apps","batch","apps.openshift.io"]
+  resources: ["jobs","deployments","replicationcontrollers","daemonsets","replicasets","statefulsets","deploymentconfigs","events","services","secrets"]
```
Comment (ksatchit) [inline deploy/rbac.yaml:22]: services & events need creation as well

File: deploy/rbac.yaml [hunk 3]
```diff
@@ -18,9 +18,15 @@ metadata:
...
+  verbs: ["get","list","watch"]
+- apiGroups: ["","litmuschaos.io"]
+  resources: ["pods","configmaps","chaosengines","chaosexperiments","chaosresults"]
+  verbs: ["get","create","update","delete","list","watch","deletecollection"]  
+- apiGroups: ["admissionregistration.k8s.io"]
+  resources: ["validatingwebhookconfigurations"]
```
Comment (ksatchit) [inline deploy/rbac.yaml:28]: lets figure out what we really need here. Operator itself doesn't use any validation webhooks. There is a separate deploy for this controller. We are adding some permissions here for that controller to work using the same svcAcc.

**Comment by uditgaurav** [comment]
#### SCENARIO TESTS

- Deleting the engine of experiment having helper pod is removing all chaos components.(abort chaos)
- kubectl apply over engine is patching the engine
- Events added by chaos-operator in engine
- A full pipeline run

**Comment by ksatchit** [comment]
Permissions looks good. Lets place the pod-delete-rbac in tests/bdd folder

**Review by ksatchit — DISMISSED** [review]

**Review by ksatchit — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Review by rahulchheda — APPROVED** [review]

**Comment by rahulchheda** [comment]
The RBAC Permissions on `rbac/deploy.yaml` and `tests/manifest/pod_delete_rbac.yaml` seems to be different, is there any purpose behind it?
cc: @ksatchit @uditgaurav

**Comment by uditgaurav** [comment]
@rahulchheda The `deploy/rbac.yaml` is for operator minimum permission and `tests/manifest/pod_delete_rbac.yaml` is the rbac of pod-delete exp as we are using pod-delete exp in bdds and we were using litmus sa for that which  we can not do now that's need pod delete a separately.

**Pull request merged**